frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx

```diff
@@ -20,14 +27,35 @@ const SilenceUntil = ({ rule }) => {
 };
 
 const SilenceAlert: React.FC<SilenceAlertProps> = ({ rule }) => {
-  const [isChecked, setIsChecked] = React.useState(rule.state !== RuleStates.Silenced);
-  const [ruleState] = React.useState(rule);
-  React.useEffect(() => setIsChecked(ruleState.state !== RuleStates.Silenced), [ruleState]);
+  const [isChecked, setIsChecked] = React.useState(true);
+  const rules = useSelector((state: RootState) => state.UI.getIn(['monitoring', 'devRules']));
+  const dispatch = useDispatch();
+  React.useEffect(() => {
+    if (rule.state === RuleStates.Silenced) {
+      setIsChecked(false);
+    }
+  }, [rule]);
 
   const handleChange = (checked: boolean) => {
     if (checked) {
       _.each(rule.silencedBy, (silence) => {
-        coFetchJSON.delete(`${alertManagerBaseURL}/api/v2/silence/${silence.id}`);
+        coFetchJSON
+          .delete(`${alertManagerBaseURL}/api/v2/silence/${silence.id}`)
+          .then(() => {
+            _.each(rule.alerts, (a) => {
+              _.omit(a, 'silencedBy');
```

Comment:
`_.omit` returns a new object rather than modifying `a`, so it looks like this doesn't do anything useful.

Reply:
@vikram-raj This definitely isn't doing anything but wasting part of a cycle to remove a field from an object that is immediately tossed up to the garbage collector. I think what you want to do is either change this to a map (instead of a for-each) or the less desirable choice is to use `delete a.silencedBy`. 

However, given the context and that `rules` is not your object (it comes from redux) you should not even be doing `a.state = <something>` as you're mutating Redux state when you do this.